lib/spack/spack/filesystem_view.py

```diff
@@ -105,7 +109,7 @@ def __init__(self, root, layout, **kwargs):
         self.projections = kwargs.get('projections', {})
 
         self.ignore_conflicts = kwargs.get("ignore_conflicts", False)
-        self.link = view_link_wrapper(self, kwargs.get("link", view_symlink))
+        self.link = ft.partial(kwargs.get("link", view_symlink), view=self)
```

Comment:
This is a minor OCD for better clarity:
```suggestion
        func = kwargs.get("link", view_symlink)
        self.link = ft.partial(func, view=self)
```